docs/proposals/odo-deploy.md

```diff
@@ -0,0 +1,152 @@
+# Support build and deployment of application container image using odo
+
+## Abstract
+Add a new command (verb) to build a production-like/slim container image for the project and deploy the built image on the target platform.
+
+There is an existing proposal for adding on outer-loop information (including support for multiple strategies for building and deploying the projects) to Devfile 2.1.0: 
+https://github.com/devfile/kubernetes-api/issues/49 
+
+It would be useful to start the design/development of a simpler version of `odo deploy` with devfile 2.0.0 that covers:
+- single build strategy - Dockerfile built using `buildah`
+- single deployment strategy - Kubernetes manifest deployed using `kubectl`.
+
+## Motivation
+`odo` is limited to development inner-loop for a project and there is no support for outer-loop - build a slim/production container image for your application and deploy the build container. It would be very useful for developers to be able to try inner-loop and then transition over to the outer-loop using odo. The outer-loop information could be provided by the application stack (devfile) to avoid developers having to worry about these aspects. 
```

Comment:
I don't think `odo deploy` should be the way for devs to deploy their apps as real workloads on production environments. Rather, this is the way they can test that their app is ready for outer-loop based deployments which will be typically done by pipelines.  Hence, the proposal is to use same Kubernetes/openshift target that they are using for development. 

If we really wanted to allow them yo specify a different target namespace or context information. I would prefer to avoid that though.

Reply:
It sounds like `odo deploy` is not intended to make use of any outer loop information in devfile if and when it becomes available.

Reply:
> Rather, this is the way they can test that their app is ready for outer-loop based deployments which will be typically done by pipelines. Hence, the proposal is to use same Kubernetes/openshift target that they are using for development.

Makes a lot of sense to me now. I know that we're working on adding `odo pipelines` support but didn't know about the whole idea. Thanks for explaining, Neeraj. :+1: